[BUG]: Win11 - Path.apply throws regex exception

### Version

0.16.2

### Scala Version

3.6.4

### Expected Behavior

This code will work on Windows 11 without exception, and information about file is shown:
```scala
import kyo.*

object Properties:
  val file: Path = Path("config.txt")

object DataApp extends KyoApp:
  run:
    for
      exists <- Properties.file.exists
      _ <- Console.printLine(exists)
    yield ()
```



### Actual Behavior

On Windows 11:
```
Exception in thread "sbt-bg-threads-1" java.lang.ExceptionInInitializerError
        at DataApp$.$init$$$anonfun$1(Main.scala:9)
        at kyo.KyoAppPlatformSpecific.v$proxy1$1(KyoAppPlatformSpecific.scala:8)
        at kyo.KyoAppPlatformSpecific.$anonfun$1$$anonfun$1(KyoAppPlatformSpecific.scala:8)
        at kyo.Async$package$Async$$anon$4.apply(Async.scala:107)
        at kyo.Async$package$Async$$anon$4.apply(Async.scala:107)
        at kyo.Async$package$Async$$anon$1.apply(Async.scala:107)
        at kyo.Abort$RunOps$$anon$1.apply(Abort.scala:177)
        at kyo.Abort$RunOps$$anon$2.apply(Abort.scala:166)
        at kyo.Abort$RunOps$$anon$1.apply(Abort.scala:177)
        at kyo.Abort$RunOps$$anon$2.apply(Abort.scala:166)
        at kyo.IO$package$IO$Unsafe$.evalLoop$1(IO.scala:113)
        at kyo.IO$package$IO$Unsafe$.evalOrThrow(IO.scala:113)
        at kyo.KyoAppPlatformSpecific.run$$anonfun$1(KyoAppPlatformSpecific.scala:8)
        at kyo.KyoApp$.kyo$KyoApp$Base$$_$main$$anonfun$1(KyoApp.scala:42)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.immutable.List.foreach(List.scala:334)
        at kyo.KyoApp$Base.main(KyoApp.scala:42)
        at DataApp.main(Main.scala)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at sbt.Run.invokeMain(Run.scala:135)
        at sbt.Run.execute$1(Run.scala:85)
        at sbt.Run.$anonfun$runWithLoader$5(Run.scala:112)
        at sbt.Run$.executeSuccess(Run.scala:178)
        at sbt.Run.runWithLoader(Run.scala:112)
        at sbt.Defaults$.$anonfun$bgRunTask$6(Defaults.scala:2072)
        at sbt.Defaults$.$anonfun$termWrapper$2(Defaults.scala:2011)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at scala.util.Try$.apply(Try.scala:213)
        at sbt.internal.BackgroundThreadPool$BackgroundRunnable.run(DefaultBackgroundJobService.scala:378)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.util.regex.PatternSyntaxException: Unescaped trailing backslash near index 1
\
        at java.base/java.util.regex.Pattern.error(Pattern.java:2045)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1804)
        at java.base/java.util.regex.Pattern.<init>(Pattern.java:1445)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1084)
        at java.base/java.lang.String.split(String.java:3161)
        at java.base/java.lang.String.split(String.java:3207)
        at kyo.Path$.apply(Path.scala:363)
        at kyo.Path$.apply(Path.scala:367)
        at Properties$.<clinit>(Main.scala:4)
        ... 34 more
```
Tested on Archlinux and same code executes without exception.

### Steps to Reproduce

Run this code:
```scala
import kyo.*

object Properties:
  val file: Path = Path("config.txt")

object DataApp extends KyoApp:
  run:
    for
      exists <- Properties.file.exists
      _ <- Console.printLine(exists)
    yield ()
```

### Current Workaround

Not known.

--- a/kyo-core/jvm/src/main/scala/kyo/Path.scala
+++ b/kyo-core/jvm/src/main/scala/kyo/Path.scala
@@ -14,6 +14,7 @@ import java.nio.file.*
 import java.nio.file.Files as JFiles
 import java.nio.file.Path as JPath
 import java.nio.file.attribute.BasicFileAttributes
+import java.util.regex.Pattern
 import kyo.*
 import kyo.Tag
 import scala.io.*
@@ -364,7 +365,7 @@ object Path:
         }
         val javaPath       = if flattened.isEmpty then Paths.get("") else Paths.get(flattened.head, flattened.tail*)
         val normalizedPath = javaPath.normalize().toString
-        if normalizedPath.isEmpty then empty else new Path(normalizedPath.split(File.separator).toList)
+        if normalizedPath.isEmpty then empty else new Path(normalizedPath.split(Pattern.quote(File.separator)).toList)
     end apply
 
     def apply(path: Part*): Path =

